### PR TITLE
Change names

### PR DESCRIPTION
**Pull request by gpescia**

### CODE REVIEW

File: netket/hilbert/__init__.py [hunk 1]
```diff
@@ -17,15 +17,14 @@
...
+from .abstract_continuous_space_particle import ContinuousHilbert
 
 from .custom_hilbert import CustomHilbert
 from .doubled_hilbert import DoubledHilbert
 from .spin import Spin
 from .fock import Fock
 from .qubit import Qubit
-from .continuous_space_boson import ContinuousBoson
-from .continuous_space_fermion import ContinuousFermion
+from .continuous_particle import ContinuousParticle
```
Comment (PhilipVinc) [inline netket/hilbert/__init__.py:27]: didn't we say particle? @gcarleo

File: netket/hilbert/continuous_particle.py [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+class ContinuousParticle(ContinuousHilbert):
+    r"""Hilbert space derived from AbstractParticle for
+    Fermions."""
+
+    def __init__(
+        self,
+        N: int,
+        L: Tuple[float, ...],
+        pbc: Union[bool, Tuple[bool, ...]],
+        mode=AnyStr,
```
Comment (PhilipVinc) [inline netket/hilbert/continuous_particle.py:15]:
```suggestion
        mode: AnyStr,
```

File: netket/hilbert/continuous_particle.py [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+        self._pbc = pbc
+        if not len(self._L) == len(self._pbc):
+            raise ValueError(
+                """`pbc` must be either a bool or a tuple indicating the periodicity of each spatial dimension."""
+            )
+
+        super().__init__(self._L)
+
+    @property
+    def mode(self) -> AnyStr:
```
Comment (PhilipVinc) [inline netket/hilbert/continuous_particle.py:44]: do we actually need this now? can we avoid adding it until we will have a use case for it?

File: netket/hilbert/continuous_particle.py [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+    @property
+    def extend(self) -> Tuple[float, ...]:
+        r"""Spatial extension in each spatial dimension"""
+        return self._L
+
+    @property
+    def pbc(self) -> Tuple[bool, ...]:
+        r"""Whether or not to use periodic boundary conditions for each spatial dimension"""
+        return self._pbc
+
```
Comment (PhilipVinc) [inline netket/hilbert/continuous_particle.py:65]: this stuff should be included in the domain object...

File: netket/hilbert/continuous_particle.py (hunk 3 above)
Comment (gcarleo) [reply netket/hilbert/continuous_particle.py:44]: yes I would remove it

File: netket/hilbert/continuous_particle.py [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+    @property
+    def n_particles(self) -> int:
+        r"""The number of particles"""
+        return self._N
+
+    @property
+    def extend(self) -> Tuple[float, ...]:
+        r"""Spatial extension in each spatial dimension"""
+        return self._L
+
```
Comment (PhilipVinc) [inline netket/hilbert/continuous_particle.py:60]:
```suggestion
```

I think this can go now?

Comment (gpescia) [reply netket/hilbert/continuous_particle.py:60]: of course sorry.

Comment (PhilipVinc) [reply netket/hilbert/continuous_particle.py:60]: you're not excused

File: netket/hilbert/__init__.py (hunk 1 above)
Comment (gcarleo) [reply netket/hilbert/__init__.py:27]: we did

File: netket/hilbert/abstract_continuous_space_particle.py [hunk 6]
```diff
@@ -3,56 +3,33 @@
...
-        return self._N
-
-    @property
-    def extend(self) -> Tuple[float, ...]:
+    def extent(self) -> Tuple[float, ...]:
         r"""Spatial extension in each spatial dimension"""
-        return self._L
+        return self._extent
 
     @property
```
Comment (PhilipVinc) [inline netket/hilbert/abstract_continuous_space_particle.py:34]: Is PBC a property for hilbert spaces?

I guess what it tells us is basically if the wavefunctions in this space are supposed to be periodic or should decay to zero...

So it kind-of look like a property of the Hilbert space.

We agree on that?

**Comment by PhilipVinc** [comment]
Can you cherry pick the commits changing name to Particle in this PR so we merge it?

File: netket/hilbert/particle.py [hunk 7]
```diff
@@ -0,0 +1,58 @@
+from typing import Tuple, Union
+
+# from netket.utils import is_scalar
+
```
Comment (PhilipVinc) [inline netket/hilbert/particle.py:4]:
```suggestion
```

**Review by PhilipVinc — APPROVED** [review]
That's very good.
Thanks @gpescia

**Pull request merged**